### PR TITLE
Prevent joining a faction with too many players in HvH gamemode

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Prevent one faction from having too many players

## Changelog
:cl:
balance: Prevent joining a faction with too many players in HvH gamemode
/:cl:

### CODE REVIEW

**Comment by ZephyrTFA** [comment]
Can you make it so you also can't join as marine in general if xenos are imbalanced? 🤡

**Comment by BraveMole** [comment]
> Can you make it so you also can't join as marine in general if xenos are imbalanced? 🤡

We discussed that earlier, this was vetoed. We don't force people onto a faction

File: code/datums/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -663,3 +661,6 @@ GLOBAL_LIST_INIT(bioscan_locations, list(
 /datum/game_mode/proc/scale_squad_jobs()
 	var/datum/job/scaled_job = SSjob.GetJobType(/datum/job/terragov/squad/leader)
 	scaled_job.total_positions = length(SSjob.active_squads[FACTION_TERRAGOV])
+
+/datum/game_mode/proc/get_joinable_factions()
```
Comment (TiviPlus) [inline code/datums/gamemodes/game_mode.dm:665]: Autodoc

Comment (BraveMole) [inline code/datums/gamemodes/game_mode.dm:665]:
```suggestion
///Return the list of joinable factions, with regards with the current round balance
/datum/game_mode/proc/get_joinable_factions()
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**